aioftp/client.py

```diff
@@ -547,13 +722,26 @@ def list(self, path="", *, recursive=False):
 
             >>> stats = await client.list()
         """
-        class AsyncClientLister(AsyncListerMixin):
+        class AsyncLister(AsyncListerMixin):
 
             async def _new_stream(cls, local_path):
 
                 cls.path = local_path
-                command = str.strip("MLSD " + str(cls.path))
-                return await self.get_stream(command, "1xx")
+                str_path = " " + str(cls.path)
+
+                cls.parse_line = self.parse_mlsx_line
+
+                try:
+                    command = str.strip("MLSD" + str_path)
+                    return await self.get_stream(command, "1xx")
+
+                except errors.StatusCodeError as e:
+
+                    if e.received_codes[-1] == "500":
```

Comment:
Use `matches` method of `Code` here. Also, not sure if it is only `500`, since `aioftp.server` sends `502 Command not implemented` on missing command. So, probably `50x` or even `5xx` is more common.